Pages/Table.php

```diff
@@ -2831,18 +2831,21 @@ function convert_quotes($v) {
         return str_replace("'", "&apos;", str_replace('"',"&quot;", $v));
     }
 
-    protected function getNewRandomImageName($field) {
-        $extension = !empty($field['extension']) ? $field['extension'] : (!empty($field['format']) ? $field['format'] : 'jpg');
+    protected function getNewRandomImageName($field, $file) {
+        $filename = $file['name'];
+        $bits = explode('.', $filename);
+        $extension = end($bits);
```

Comment:
this seems to always take the same extension as the original image, but that's not always true. for example if the table preset says it has ot be a jpg, then it will be jpg. if they upload a png it will have to be converted. so the original format will have the wrong extension